tests/testthat/test_diffexp_seq.R

```diff
@@ -0,0 +1,154 @@
+                                    pair_id = "Pair", pair_group = "Pair_group", pair_denom = "Potato_1")
+  
+  ## pairs only
+  seqdata_grp8 <- group_designation(seqdata,
+                                    pair_id = "Pair", pair_group = "Pair_group", pair_denom = "Potato_1")
+  
+  
+  list_of_omics <- list(seqdata_grp, #seqdata_grp2, 
+                        seqdata_grp3, seqdata_grp4,
+                        seqdata_grp5, seqdata_grp6,
+                        seqdata_grp7, seqdata_grp8
+                        )
+  
+  #### Argument errors ####
+  
+  expect_error(diffexp_seq(seqdata_grp, method = "fhk"),
+               "method must a single character string of length 1 in 'edgeR', 'DESeq2', or 'voom'")
+  
+  expect_error(diffexp_seq(seqdata_grp$f_data, method = "edgeR"), 
+               "omicsData must be of class 'seqData'")
+  
+  expect_error(diffexp_seq(seqdata_grp, p_adjust = "tiger"), 
+               "p_adjust must a single character string of")
+  
+  expect_error(diffexp_seq(seqdata_grp, p_cutoff = "tiger"), 
+               "p_cutoff must numeric of length 1")
+  
+  # Custom comparisons ---------------
+  comparisons <- data.frame(
+    Test = "uterus_hCG",
+    Control = "uterus_PBS"
+  )
+  
+  bad_comparisons <- data.frame(
+    Test = "uterus_hC",
+    Control = "uterus_PBS"
+  )
+
+  err_comps <- "Invalid comparisons given"
+  res <- purrr::map(c("edgeR", "DESeq2", "voom"), function(method){
+    expect_error(diffexp_seq(seqdata_grp, method = method, comparisons = bad_comparisons), err_comps)
+  })
+  
+  ## Ok versions ##
+  run_all <- purrr::map(c("edgeR", "DESeq2", "voom"), function(method){
+    
+    ## basic
+    basic_list <- purrr::map(1:length(list_of_omics), function(n){
+      
+      omics <- list_of_omics[[n]]
+      
+      ## warnings tested in get_group_formula as well as minor warnings:
+      # character -> factor, rowname/colname coef magic
+      suppressWarnings(diffexp_seq(omics, method = method))
+    })
+    
+    ## pvalue adjust
+    p_adjust_none <-  list(diffexp_seq(seqdata_grp, method = method, p_adjust = "none"))
+
+    p_adjust_holm <- list(diffexp_seq(seqdata_grp, method = method, p_adjust = "holm"))
+    
+    ## custom list
+    comps <- list(diffexp_seq(seqdata_grp, method = method, comparisons = comparisons))
+    
+    
+    return(c(basic_list, p_adjust_none, p_adjust_holm, comps))
+  })
+  
+  names(run_all) <- c("edgeR", "DESeq2", "voom")
+  
+
+  ## check IDs on all dfs for consistency with edata
+  res <- purrr::map(run_all, function(method_res){
+    purrr::map(method_res, function(df){
+      expect_true(all(df$ID_REF == seqdata_grp$e_data$ID_REF))
+    })
+  })
+  
+  
+  ## Check dimensions
+  
+  
+  ## Check with previously saved values
+  
+  ## check using ... in calls
+  fitType <- "parametric"
+  fittype_change <- diffexp_seq(seqdata_grp, method = "DESeq2", fitType = fitType)
```

Comment:
Cant tell if these are missing something, or should be deleted?

Reply:
@clabornd so fitType isn't explicitly defined by diffexp_seq but is acceptable in the DESeq2 functions -- this line is to mostly ensure that the passing from "..." doesn't error out innately